File: mongodb-api/src/main/java/com/github/mehdishahdoost/mongodbapi/controller/StudentController.java
package com.github.mehdishahdoost.mongodbapi.controller;


import com.github.mehdishahdoost.mongodbapi.model.Student;
import com.github.mehdishahdoost.mongodbapi.service.StudentService;
import lombok.AllArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@AllArgsConstructor
public class StudentController {

    private StudentService studentService;

    @GetMapping("/student")
    public List<Student> getStudentList() {
        return studentService.getStudents();
    }

}

File: spring-data/mongodb-api-kotlin/src/main/kotlin/com/github/mehdishahdoost/controller/StudentController.kt
package com.github.mehdishahdoost.controller

import com.github.mehdishahdoost.model.Student
import com.github.mehdishahdoost.service.StudentService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class StudentController @Autowired constructor(val studentService: StudentService) {

    @GetMapping("/students")
    fun getStudent(): List<Student> = studentService.getStudent()
}